### PR TITLE
[ci-app] Add More CIs and Improve Testing

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What and why?

Improve support for CI providers and make sure we follow the spec.

### How?

* Add JSON files for each CI provider to assert we are extracting metadata correctly. This is based on the CI Spec in https://github.com/DataDog/datadog-ci-spec/tree/main/spec_tests. 
* Create tests for each file: https://github.com/DataDog/datadog-ci/pull/219/files#diff-bdab68b0d54fe06587bd7486f7b20e4da6d3ec0127f20806fa0096546de554e6R144-R148 
* Add support for remaining CI providers in `src/helpers/ci.ts`.

Additionally I've added this repository to our spec tests in https://github.com/DataDog/datadog-ci-spec/pull/95. 

The GHA in https://github.com/DataDog/datadog-ci-spec/blob/main/.github/workflows/spec_tests.yml checks whether the projects that are reading from CI providers' environment are homogeneous. 

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "prettier-check": "prettier --check \"src/**/*.{ts,js,json,yml}\" --ignore-path .gitignore || (echo \"\nYou can fix this by running ==> yarn prettier-write <==\n\" && false)",
     "prettier-write": "prettier --write \"src/**/*.{ts,js,json,yml}\" --ignore-path .gitignore",
     "test": "jest",
-    "test:debug": "node --inspect-brk `which jest`",
+    "test:debug": "node --inspect-brk `which jest` --runInBand",
```
Comment (juan-fernandez) [inline package.json:35]: unrelated change: `--inspect-brk` does nothing if you're running tests in parallel

**Review by yoannmoinet — APPROVED** [review]
Infra LGTM.

I think you should add your team as CODEOWNER of the command and related folders.
I don't feel comfortable validating this code when I don't fully understand what it covers exactly.
Someone from your team will be more suited for this.

**Comment by juan-fernandez** [comment]
> Infra LGTM.
> 
> I think you should add your team as CODEOWNER of the command and related folders.
> I don't feel comfortable validating this code when I don't fully understand what it covers exactly.
> Someone from your team will be more suited for this.

yeah that makes sense. I'll do it 👍

**Comment by juan-fernandez** [comment]
> > Infra LGTM.
> > I think you should add your team as CODEOWNER of the command and related folders.
> > I don't feel comfortable validating this code when I don't fully understand what it covers exactly.
> > Someone from your team will be more suited for this.
> 
> yeah that makes sense. I'll do it 👍

https://github.com/DataDog/datadog-ci/pull/220

**Review by AdrianLC — APPROVED** [review]
